### DISCUSSION TITLE
ollama-api openai compatibility

### DISCUSSION THREAD

**Discussion by gipde** [open-webui/open-webui#3318] [General]
# Bug Report

## Description

**Bug Summary:**
if i use continue.dev i would like to use embeddings via open-webui like tabautocompletion or chat.

**Steps to Reproduce:**
configuration in continue.dev

```
"embeddingsProvider": {
    "model": "nomic-embed-text",
    "provider": "openai",
    "apiBase": "http://<myIP-of-open-webui>/ollama/api",
    "apiKey": "sk-"
  }
```

due to recordings, continue.dev transmit the following json fragment

```
{
    "model": "nomic-embed-text",
    "input":["hello ..."]
  }
```

but the api requires json in form of 

```
{
  "model": "nomic-embed-text",
  "prompt": "string"
}
```

so it seems that there is a incompatibility within the json structure.

on the oter hand, if i configure the ollama-provider in continue.dev, i cannot use the api-key from open-webui

**Comment by Garethp** [comment]
This would be a fantastic feature. I'd love to be able to block off the 11434 port on the machine running Ollama so that only authenticated requests get forwarded on, but without the embeddings API being supported that's not really an option

**Comment by Garethp** [comment]
There's still an issue with the embedding compatibility that I reported in #5704. @tjbck marked the issue as closed despite open-webui's api still not being properly compatible with ollama's

**Reply by tjbck** [reply]
```py
import requests

# Define the URL and headers
url = "http://localhost:8080/ollama/api/embed"
token = "token"
headers = {
    'accept': 'application/json',
    'Authorization': f'Bearer {token}',
    'Content-Type': 'application/json'
}

# Define the payload
data = {
    "model": "nomic-embed-text",
    "input": "test"
}

# Make the POST request
response = requests.post(url, headers=headers, json=data)

# Print the response
print(response.status_code)
print(response.json())
```

Works with the latest, I cannot reproduce your issue. Please try again with the latest version, and let me know.

**Reply by Garethp** [reply]
Thanks for the reploy @tjbck. I've retested on open-webui version v0.3.32 with ollama v0.3.13 and running

```sh
curl -X POST https://open-web-ui/ollama/api/embed \
	-H 'accept: application/json' \
	-H 'Authorization: Bearer jwt-token' \
	-H 'Content-Type: application/json' \
	-d '{
		"model": "nomic-embed-text",
		"input": "test"
	}'
```

Still outputs the error:

```json
{"detail":[{"type":"missing","loc":["body","truncate"],"msg":"Field required","input":{"model":"nomic-embed-text","input":"test"}}]}                                                   
```

I'm deploying the `open-webui:main` docker image for my testing

**Reply by Garethp** [reply]
@tjbck, did you have any luck replicating this in the docker version?
